cleanrl/sac_continuous_action_jax.py

```diff
@@ -0,0 +1,548 @@
+            out_axes=0,
+            axis_size=self.n_critics,
+        )
+        q_values = vmap_critic(
+            n_units=self.n_units,
+        )(obs, action)
+        return q_values
+
+
+class Actor(nn.Module):
+    action_dim: Sequence[int]
+    n_units: int = 256
+    log_std_min: float = -20
+    log_std_max: float = 2
+
+    @nn.compact
+    def __call__(self, x: jnp.ndarray) -> tfd.Distribution:
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        mean = nn.Dense(self.action_dim)(x)
+        log_std = nn.Dense(self.action_dim)(x)
+        log_std = jnp.clip(log_std, self.log_std_min, self.log_std_max)
+        dist = TanhTransformedDistribution(
+            tfd.MultivariateNormalDiag(loc=mean, scale_diag=jnp.exp(log_std)),
+        )
+        return dist
+
+
+@partial(jax.jit, static_argnames="actor")
+def sample_action(
+    actor: Actor,
+    actor_state: TrainState,
+    observations: jnp.ndarray,
+    key: jax.random.KeyArray,
+) -> jnp.array:
+    dist = actor.apply(actor_state.params, observations)
+    action = dist.sample(seed=key)
+    return action
+
+
+@partial(jax.jit, static_argnames="actor")
+def select_action(actor: Actor, actor_state: TrainState, observations: jnp.ndarray) -> jnp.array:
+    return actor.apply(actor_state.params, observations).mode()
+
+
+def scale_action(action_space: gym.spaces.Box, action: np.ndarray) -> np.ndarray:
+    """
+    Rescale the action from [low, high] to [-1, 1]
+    (no need for symmetric action space)
+
+    :param action: Action to scale
+    :return: Scaled action
+    """
+    low, high = action_space.low, action_space.high
+    return 2.0 * ((action - low) / (high - low)) - 1.0
+
+
+def unscale_action(action_space: gym.spaces.Box, scaled_action: np.ndarray) -> np.ndarray:
+    """
+    Rescale the action from [-1, 1] to [low, high]
+    (no need for symmetric action space)
+
+    :param scaled_action: Action to un-scale
+    """
+    low, high = action_space.low, action_space.high
+    return low + (0.5 * (scaled_action + 1.0) * (high - low))
+
+
+@dataclass
+class SB3Adapter:
+    """
+    Adapter in order to use ``evaluate_policy()`` from Stable-Baselines3.
+    """
+
+    actor: Actor
+    actor_state: RLTrainState
+    key: jax.random.KeyArray
+    action_space: gym.spaces.Box
+
+    def predict(self, observations: np.ndarray, deterministic=True, state=None, episode_start=None):
+        if deterministic:
+            actions = select_action(self.actor, self.actor_state, observations)
+        else:
+            self.key, noise_key = jax.random.split(self.key, 2)
+            actions = sample_action(self.actor, self.actor_state, observations, noise_key)
+
+        # Clip due to numerical instability
+        actions = np.clip(actions, -1, 1)
+        # Rescale to proper domain when using squashing
+        actions = unscale_action(self.action_space, actions)
+
+        return actions, None
```

Comment:
I will need to think of a better way to handle evaluation, probably after #292 is merged. It's good to keep it as is for now.